src/Compilers/Core/Portable/SourceGeneration/Nodes/SyntaxValueProvider_ForAttributeWithSimpleName.cs

```diff
@@ -338,29 +338,34 @@ bool matchesAttributeName(string currentAttributeName, bool withAttributeSuffix)
                 return false;
 
             seenNames.Push(currentAttributeName);
-
-            foreach (var (aliasName, symbolName) in localAliases)
+            try
             {
-                // see if user wrote `[SomeAlias]`.  If so, if we find a `using SomeAlias = ...` recurse using the
-                // ... name portion to see if it might bind to the attr name the caller is searching for.
-                if (matchesName(currentAttributeName, aliasName, withAttributeSuffix) &&
-                    matchesAttributeName(symbolName, withAttributeSuffix: false))
+                foreach (var (aliasName, symbolName) in localAliases)
                 {
-                    return true;
+                    // see if user wrote `[SomeAlias]`.  If so, if we find a `using SomeAlias = ...` recurse using the
+                    // ... name portion to see if it might bind to the attr name the caller is searching for.
+                    if (matchesName(currentAttributeName, aliasName, withAttributeSuffix) &&
+                        matchesAttributeName(symbolName, withAttributeSuffix: false))
+                    {
+                        return true;
+                    }
                 }
-            }
 
-            foreach (var (aliasName, symbolName) in globalAliases.AliasAndSymbolNames)
-            {
-                if (matchesName(currentAttributeName, aliasName, withAttributeSuffix) &&
-                    matchesAttributeName(symbolName, withAttributeSuffix: false))
+                foreach (var (aliasName, symbolName) in globalAliases.AliasAndSymbolNames)
                 {
-                    return true;
+                    if (matchesName(currentAttributeName, aliasName, withAttributeSuffix) &&
+                        matchesAttributeName(symbolName, withAttributeSuffix: false))
+                    {
+                        return true;
+                    }
                 }
-            }
 
-            seenNames.Pop();
-            return false;
+                return false;
+            }
+            finally
+            {
+                seenNames.Pop();
```

Comment:
early exits in the foreach code (a `return true;`) was causing us to not pop the vlaues we'd pushed onto this stack, leaving it in an inconsistent state.

In the test case this manifested as us going "we've already seen an attribute with this alias name, so we shouldn't look at it again" when we hit the second attribute.  

I considered changing to have a single value that is retruned at the end, after hte pop.  but i prefer this as it is more future proof, and clearly ensures that any sort of exiting will always leave things consistent.